crates/red_knot_python_semantic/resources/mdtest/attributes.md

```diff
@@ -295,14 +318,50 @@ class C:
         for _, self.y in TupleIterable():
             pass
 
-# TODO: Pyright fully supports these, mypy detects the presence of the attributes,
-# but infers type `Any` for both of them. We should infer `int` and `str` here:
+reveal_type(C().x)  # revealed: Unknown | int
+
+reveal_type(C().y)  # revealed: Unknown | str
```

Comment:
The TODO message was not quite right here, I think `Unknown | int` / `Unknown | str` are the correct types here. There is no declaration.

The attributes could technically also be unbound. We don't discover this in other cases either, though.

Reply:
Yes, determining that an attribute is definitely-initialized is a complex problem that I think we should defer for now.